Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -31,8 +31,92 @@
 #import <algorithm>
 #import <vector>
 
+#import <CoreFoundation/CoreFoundation.h>
+#import <CFRuntime.h>
+#import "D2DWrapper.h"
+
+#include <COMIncludes.h>
+#import <wrl/client.h>
+#import <D2d1.h>
+#include <COMIncludes_End.h>
+
 static const wchar_t* TAG = L"CGPath";
 
+using namespace std;
+using namespace Microsoft::WRL;
+
+struct __CGPath {
+    CFRuntimeBase _base;
+
+    ComPtr<ID2D1PathGeometry> _pathGeometry;
+    ComPtr<ID2D1GeometrySink> _geometrySink;
+
+    bool _isPathClosed;
+    CGPoint _currentPoint;
+    CGPoint _startingPoint;
+};
+
+// A private helper function for re-opening a path geometry.
+static void __preparePathForEditing(CGPathRef path) {
```

Comment:
It seems to destroy the existing path (as both _pathGeometry and _geometrySink are recreated?  Also, we are leaking the old objects.

---
In reply to: [83128267](https://github.com/Microsoft/WinObjC/pull/1138#discussion_r83128267) [](ancestors = 83128267)